pkg/autoupdate/tuf/library_lock.go

```diff
@@ -0,0 +1,32 @@
+package tuf
+
+import "sync"
+
+// libraryLock wraps a number of locks, ensuring that any one binary's library
+// can only be modified by one routine at a time.
+type libraryLock struct {
```

Comment:
Why not have this on librarian directly?

Reply:
It seemed tidier to have it abstracted out, but that might have just been because I'd gone through other refactoring to split out the autoupdater from the librarian recently!